.devops/deploy-pipelines.yml

```
@@ -9,18 +9,41 @@ parameters:
       - minor
       - patch
     default: minor
+  - name: environment
+    displayName: Target Environment
+    type: string
+    default: uat
+    values:
+      - uat
+      - prod
 
 # Only manual activations are intended
 trigger: none
 pr: none
 
-# This pipeline has been implemented to be run on hosted agent pools based both
-# on 'windows' and 'ubuntu' virtual machine images and using the scripts defined
-# in the package.json file. Since we are deploying on Azure functions on Windows
-# runtime, the pipeline is currently configured to use a Windows hosted image for
-# the build and deploy.
+# variables setted on azdevops to avoid circular dependency warning
+# $(DEFAULT_BRANCH)
+# $(HEALTHCHECK_CONTAINER_SUBNET)
+variables:
+  ${{ if eq(parameters['environment'], 'uat') }}:
+    DEPLOY_TYPE: $[variables['UAT_DEPLOY_TYPE']]
+    AZURE_SUBSCRIPTION: $(UAT_AZURE_SUBSCRIPTION)
+    FUNCTION_APP_NAME: $(UAT_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(UAT_WEB_APP_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_ENDPOINT: $(UAT_HEALTHCHECK_ENDPOINT)
+    HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME: $(UAT_HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_CONTAINER_VNET: $(UAT_HEALTHCHECK_CONTAINER_VNET)
+  ${{ if eq(parameters['environment'], 'prod') }}:
+    DEPLOY_TYPE: $[variables['PROD_DEPLOY_TYPE']]
+    AZURE_SUBSCRIPTION: $(PROD_AZURE_SUBSCRIPTION)
+    FUNCTION_APP_NAME: $(PROD_FUNCTION_APP_NAME)
+    FUNCTION_APP_RESOURCE_GROUP_NAME: $(PROD_WEB_APP_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_ENDPOINT: $(PROD_HEALTHCHECK_ENDPOINT)
+    HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME: $(PROD_HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME)
+    HEALTHCHECK_CONTAINER_VNET: $(PROD_HEALTHCHECK_CONTAINER_VNET)
+
 pool:
-  vmImage: 'windows-2019'
+  vmImage: 'ubuntu-latest'
```

```suggestion
  vmImage: 'windows-2019'
```
We use windows function-app please use windows image machine